File: frontend/src/types/playbook.ts
export type PlaybookAction = {
  action: string;
  reason?: string;
  priority: "high" | "medium" | "low";
};

export type PlaybookJSON = {
  what_happened: string;
  associated_risk: string;
  recommended_actions: PlaybookAction[];
};

export type PlaybookSource = {
  id?: number | string;
  source?: string;
  text?: string;
  score?: number;
};

export type PlaybookRecord = {
  playbook_id: number;
  alert_id: string;
  created_at: string;
  model: string;
  playbook: PlaybookJSON;
  sources: PlaybookSource[];
  confidence: "high" | "medium" | "low";
};

File: src/backend/app/schemas/playbook.py
from pydantic import BaseModel, Field
from typing import List, Optional, Literal, Dict, Any
from datetime import datetime

Priority = Literal["high", "medium", "low"]

class RecommendedAction(BaseModel):
    action: str
    reason: str
    priority: Priority = "medium"

class PlaybookJSON(BaseModel):
    what_happened: str = Field(..., description="Summary of what happened based on alert + context")
    associated_risk: str = Field(..., description="Risk/impact if this is true positive")
    recommended_actions: List[RecommendedAction] = Field(..., description="Ordered list of actions")

class PlaybookRecord(BaseModel):
    playbook_id: str
    alert_id: str
    created_at: datetime
    model: str
    playbook: PlaybookJSON
    sources: List[Dict[str, Any]] = []
    confidence: str = "medium"
